src/components/KDropdown/KDropdownItem.vue

```diff
@@ -0,0 +1,296 @@
+}>()
+
+const type = computed((): DropdownItemType => {
+  if (props.item?.to) {
+    return 'link'
+  } else if (typeof props.onClick !== 'undefined' || props.selectionMenuChild) {
+    // checking attrs since we deleted click from listeners
+    return 'button'
+  }
+  return 'default'
+})
+
+const label = computed((): string => (props.item?.label) || '')
+
+const to = computed((): string | object | undefined => (props.item?.to) || undefined)
+
+const handleClick = (event: Event): void => {
+  emit('click', event)
+
+  if (props.selectionMenuChild) {
+    emit('change', props.item)
+  }
+}
+
+const componentType = computed((): DropdownItemRenderedType => {
+  let result: DropdownItemRenderedType = 'div'
+
+  if (type.value === 'link' && to.value && typeof to.value === 'string') {
+    result = 'link'
+  } else if (type.value === 'link' && to.value && typeof to.value === 'object') {
+    result = 'router-link'
+  } else if (type.value === 'button') {
+    result = 'button'
+  }
+
+  return result
+})
+
+// Strips falsy `disabled` attribute, so it does not fall onto native <a> elements.
+const strippedAttrs = computed((): typeof attrs => {
+  const modifiedAttrs = Object.assign({}, attrs)
+
+  delete modifiedAttrs.class
+  delete modifiedAttrs.disabled
+
+  return modifiedAttrs
+})
+
+const availableComponents = computed((): DropdownItemRenderedRecord => ({
+  link: {
+    tag: 'a',
+    attrs: {
+      class: `dropdown-item-trigger ${props.disabled ? 'disabled' : ''} ${attrs.class || ''}`,
+      href: to.value as string,
+      // only add disabled attribute if props.disabled returns truthy value, otherwise it will be added as disabled="false" which will be treaded as disabled
+      ...(!!props.disabled && { disabled: true, tabindex: -1 }),
+      ...strippedAttrs.value, // TODO: mention this in the docs
+    },
+  },
+  'router-link': {
+    tag: 'router-link',
+    onClick: handleClick,
+    attrs: {
+      class: `dropdown-item-trigger ${props.disabled ? 'disabled' : ''} ${attrs.class || ''}`,
+      to: to.value,
+      // only add disabled attribute if props.disabled returns truthy value, otherwise it will be added as disabled="false" which will be treaded as disabled
+      ...(!!props.disabled && { disabled: true, tabindex: -1 }),
+      ...strippedAttrs.value,
+    },
+  },
+  button: {
+    tag: 'button',
+    onClick: handleClick,
+    attrs: {
+      class: `dropdown-item-trigger ${props.disabled ? 'disabled' : ''} ${attrs.class || ''}`,
+      disabled: props.disabled,
+      ...strippedAttrs.value,
+    },
+  },
+  div: {
+    tag: 'div',
+    attrs: {
+      class: 'dropdown-item-trigger',
+    },
+  },
+}))
+</script>
+
+<style lang="scss" scoped>
+/* Component mixins */
+
+@mixin kDropdownItemTriggerDefaults {
+  transition: background-color $kongponentsTransitionDurTimingFunc, color $kongponentsTransitionDurTimingFunc;
+}
+
+/* Component styles */
+
+.k-dropdown-item, :deep(.k-dropdown-item) {
```

Comment:
The hover color on a dropdown item is not even noticeable -- it needs to be darker
![image](https://github.com/Kong/kongponents/assets/2229946/2b878530-2ddc-4d51-a4ad-1d473aaf8d7d)

Reply:
Jilson had the same feedback will point it out to Jason 😅